cpp/oneapi/dal/io/csv/common.hpp

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/detail/common.hpp"
+#include "oneapi/dal/table/common.hpp"
+
+namespace oneapi::dal::csv {
+
+namespace detail {
+struct data_source_tag {};
+class data_source_impl;
+} // namespace detail
+
+class ONEAPI_DAL_EXPORT data_source_base : public base {
```

Comment:
Why do you need `data_source_base`? Asking because this technique applied with templated descriptors, but `data_source` has no template params

Reply:
Removed _base one.